### PR TITLE
*: update TiDB product names for cloud-console content

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

This PR applies the new product names (TiDB Self-Managed, TiDB Cloud Serverless, TiDB Cloud Dedicated) to files in the `/tidb-cloud` directory and the file `TOC-tidb-cloud.md`. This PR is related to https://github.com/pingcap/docs/pull/18744.

Note: The `tidb-cloud/dev-guide-bi-looker-studio.md` file is also updated in this PR to pass the link check. And I'll handle the possible conflict when merging these two PRs (https://github.com/pingcap/docs/pull/18744 and this PR).

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs/pull/18744

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/dev-guide-bi-looker-studio.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ To complete this tutorial, you need:
 
 **If you don't have a TiDB Cloud Serverless cluster, you can create one as follows:**
 
-- [Create a TiDB Cloud Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md#step-1-create-a-tidb-cloud-serverless-cluster)
+- [Create a TiDB Cloud Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md#step-1-create-a-tidb-serverless-cluster)
```
Comment (lilin90) [inline tidb-cloud/dev-guide-bi-looker-studio.md:28]: Note: I'll update this anchor link by adding `cloud` in https://github.com/pingcap/docs/pull/18744.

File: tidb-cloud/serverless-driver-drizzle-example.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ summary: Learn how to use TiDB Cloud serverless driver with Drizzle.
 
 # TiDB Cloud Serverless Driver Drizzle Tutorial
 
-[Drizzle ORM](https://orm.drizzle.team/) is a lightweight and performant TypeScript ORM with developer experience in mind. Starting from `drizzle-orm@0.31.2`, it supports [drizzle-orm/tidb-serverless](https://orm.drizzle.team/docs/get-started-mysql#tidb-serverless), enabling you to use Drizzle over HTTPS with [TiDB Cloud serverless driver](/tidb-cloud/serverless-driver.md).
+[Drizzle ORM](https://orm.drizzle.team/) is a lightweight and performant TypeScript ORM with developer experience in mind. Starting from `drizzle-orm@0.31.2`, it supports [drizzle-orm/tidb-serverless](https://orm.drizzle.team/docs/get-started-mysql#tidb-cloud-serverless), enabling you to use Drizzle over HTTPS with [TiDB Cloud serverless driver](/tidb-cloud/serverless-driver.md).
```
Comment (qiancai) [inline tidb-cloud/serverless-driver-drizzle-example.md:8]:
```suggestion
[Drizzle ORM](https://orm.drizzle.team/) is a lightweight and performant TypeScript ORM with developer experience in mind. Starting from `drizzle-orm@0.31.2`, it supports [drizzle-orm/tidb-serverless](https://orm.drizzle.team/docs/get-started-mysql#tidb-serverless), enabling you to use Drizzle over HTTPS with [TiDB Cloud serverless driver](/tidb-cloud/serverless-driver.md).
```

File: tidb-cloud/serverless-driver-drizzle-example.md [hunk 3]
```diff
@@ -268,5 +268,5 @@ To complete this tutorial, you need the following:
 
 ## What's next
 
-- Learn more about [Drizzle](https://orm.drizzle.team/docs/overview) and [drizzle-orm/tidb-serverless](https://orm.drizzle.team/docs/get-started-mysql#tidb-serverless).
+- Learn more about [Drizzle](https://orm.drizzle.team/docs/overview) and [drizzle-orm/tidb-serverless](https://orm.drizzle.team/docs/get-started-mysql#tidb-cloud-serverless).
```
Comment (qiancai) [inline tidb-cloud/serverless-driver-drizzle-example.md:271]:
```suggestion
- Learn more about [Drizzle](https://orm.drizzle.team/docs/overview) and [drizzle-orm/tidb-serverless](https://orm.drizzle.team/docs/get-started-mysql#tidb-serverless).
```

File: tidb-cloud/release-notes-2022.md
Comment (qiancai) [inline tidb-cloud/release-notes-2022.md:1]: For previously published release notes, could we retain the old product names?

File: tidb-cloud/notification-2024-04-09-monitoring-features-maintenance.md
Comment (qiancai) [inline tidb-cloud/notification-2024-04-09-monitoring-features-maintenance.md:1]: For previously published notifications, could we retain the old product names?

File: tidb-cloud/dev-guide-bi-looker-studio.md (hunk 1 above)
Comment (lilin90) [reply tidb-cloud/dev-guide-bi-looker-studio.md:28]: Updated in https://github.com/pingcap/docs/pull/18744/files#diff-b24124ffe092ce9c9a30a0faf3bb94a789e2fb05999dd6c492e60c5f3b20a1d9.

File: tidb-cloud/notification-2024-04-09-monitoring-features-maintenance.md
Comment (lilin90) [reply tidb-cloud/notification-2024-04-09-monitoring-features-maintenance.md:1]: Or just use the new names to avoid making new users confused.

**Review by qiancai — APPROVED** [review]

**Pull request merged**